### PR TITLE
[16.0][MIG] purchase_variant_configurator_on_confirm: Merged into purchase_variant_configurator and migration to 16.0

### PR DESCRIPTION
**Pull request by victoralmau**
Merged into `purchase_variant_configurator` and migration to 16.0

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT46597

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Seems to work fine. 

An expanded readme with at least a functional testing case would be welcome

**Comment by pedrobaeza** [comment]
In fact, I think this feature should be integrated into `purchase_variant_configurator_on_confirm` as an option to configure at system or company level.

**Comment by victoralmau** [comment]
> In fact, I think this feature should be integrated into `purchase_variant_configurator_on_confirm` as an option to configure at system or company level.

I guess you mean to add this functionality to `purchase_variant_configurator` with a new field in `res.config.settings`, right?

What is the way to do it? All in the migration commit?

**Comment by pedrobaeza** [comment]
Yeah, that's it, sorry. You can do it in the migration commit, yes.

**Comment by victoralmau** [comment]
Preparing to move everything to `purchase_variant_configurator` I'm not sure that's the best decision. We want to add this https://github.com/OCA/product-variant/blob/13.0/purchase_variant_configurator_on_confirm/models/purchase_order.py#L41 and the _sql_constraints to **everyone**? I think not, and maybe for that reason it was left in a separate module.

**Comment by pedrobaeza** [comment]
I think it can be removed at DB level, and enforced in screen depending on the parameter.

**Comment by victoralmau** [comment]
Changes done.

File: purchase_variant_configurator/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            "Forbidden values on non-accountable purchase order line",
+        ),
+    ]
+
+    @api.depends("company_id")
+    def _compute_product_id_is_required(self):
+        for item in self:
+            item.product_id_is_required = (
+                False if item.company_id.po_confirm_create_variant else True
+            )
```
Comment (pedrobaeza) [inline purchase_variant_configurator/models/purchase_order.py]:
```suggestion
            item.product_id_is_required = (
                not item.company_id.po_confirm_create_variant
            )
```

File: purchase_variant_configurator/readme/CONFIGURE.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+#. Go to ``Purchase > Configuration > Settings`` and check the "Auto-create product variants" option.
```
Comment (pedrobaeza) [inline purchase_variant_configurator/readme/CONFIGURE.rst]: With the rest of the changes:

```suggestion
#. Go to ``Purchase > Configuration > Settings``.
#. Check the "Create variants on confirm" option if you want to delay the creation 
   of them until the order confirmation.
```

File: purchase_variant_configurator/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Only if the "Auto-create product variants" option in the configuration is set.
```
Comment (pedrobaeza) [inline purchase_variant_configurator/readme/USAGE.rst]: Change this according the other things

File: purchase_variant_configurator/views/res_config_settings_views.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="res_config_settings_view_form_purchase" model="ir.ui.view">
+        <field name="model">res.config.settings</field>
+        <field
+            name="inherit_id"
+            ref="purchase.res_config_settings_view_form_purchase"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@id='auto_receipt_reminder']" position="after">
```
Comment (pedrobaeza) [inline purchase_variant_configurator/views/res_config_settings_views.xml]: Please put a new section "Product configurator"

**Comment by victoralmau** [comment]
Changes done.

File: purchase_variant_configurator/models/res_company.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    po_confirm_create_variant = fields.Boolean(
+        string="Auto-create product variants",
```
Comment (pedrobaeza) [inline purchase_variant_configurator/models/res_company.py]: This text is not coherent and not changed according https://github.com/OCA/product-variant/pull/362#discussion_r1657599342

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major

**Comment by pedrobaeza** [comment]
Please add the corresponding merging in OpenUpgrade.

**Pull request merged**